internal/backend/svc/svc.go

```diff
@@ -258,8 +260,10 @@ func makeFxOpts(cfg *Config, opts RunOptions) []fx.Option {
 						varsv1connect.VarsServiceName,
 					},
 					[]httpsvc.RequestLogExtractor{
+						// Note: auth middleware is connected after interceptor, so in httpsvc and interceptor there is no user in the context.
+						// Therefore we just extract and parse user from the Authorization header.
 						func(r *http.Request) []zap.Field {
-							if user := authcontext.GetAuthnUser(r.Context()); user.IsValid() {
+							if user := authHdrExtractor(r); user.IsValid() {
```

Comment:
Please add a brief explanation why extracting the user ID from the request is needed, and extracting it from the context isn't, i.e. why the user ID can't be in the context.

Are there situations where you'll want to rely on both?